lib/engine/game/g_1862/step/merge.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1862
+      module Step
+        class Merge < Engine::Step::Base
+          def actions(entity)
+            return [] if !entity.corporation? || entity != current_entity
+            return [] if entity.receivership? || @game.skip_round[entity]
+
+            return ['choose'] if @merging
+
+            %w[merge pass]
+          end
+
+          def log_skip(entity)
+            super unless @game.skip_round[entity]
+          end
+
+          def auto_actions(entity)
+            return super if @merging
+
+            return [Engine::Action::Pass.new(entity)] if mergeable_candidates(entity).empty?
+
+            super
+          end
+
+          def merge_name
+            'Merge'
+          end
+
+          def merger_auto_pass_entity
+            current_entity
+          end
+
+          def can_merge?(entity)
+            mergeable_candidates(entity).any?
+          end
+
+          def description
+            return 'Choose Survivor' if @merging
+
+            'Merge with Major Corporation'
+          end
+
+          def process_merge(action)
+            @merging = [action.entity, action.corporation]
+            @log << "#{@merging.first.name} and #{@merging.last.name} will merge"
+          end
+
+          def process_choose(action)
+            if action.choice == :first
```

Comment:
Is it worth breaking out the choice logic into a helper function, so you don't have to duplicate in acquire?

Reply:
The only difference is if :merge or :acquire is passed in, right?

Reply:
FWIW it can be shortened into just two lines (https://github.com/tobymao/18xx/pull/5206#discussion_r625365139). Might still be worth putting those two lines into a helper function though.

Reply:
Ok.